### PR TITLE
Alpine support

### PR DESCRIPTION
**Pull request by ilpianista**
- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [ ] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Support for [Alpine Linux](https://alpinelinux.org/) 3.10.

**How does this PR accomplish the above?:**

I got pihole working on Alpine 3.10 by starting from scratch. My work has also been inspired by @deftdawg patch (#2950).

I ported away the `basic-install.sh` script from the commands not available under Alpine. This MR also adds support for the OpenRC init system.

**What documentation changes (if any) are needed to support this PR?:**

---

There's a little annoyance the first time you execute the install script because of a [bug with ip -oneline](https://gitlab.alpinelinux.org/alpine/aports/issues/9966) on Alpine 3.9 and 3.10, but it's already patched on 3.11.

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -58,7 +58,7 @@ webroot="/var/www/html"
 # We store several other directories and
 webInterfaceGitUrl="https://github.com/pi-hole/AdminLTE.git"
 webInterfaceDir="${webroot}/admin"
-piholeGitUrl="https://github.com/pi-hole/pi-hole.git"
+piholeGitUrl="https://github.com/ilpianista/pi-hole.git"
```
Comment (ilpianista) [inline automated install/basic-install.sh:61]: To be reverted before merging.

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -2227,6 +2227,7 @@ clone_or_update_repos() {
         { printf "  %bUnable to clone %s into %s, unable to continue%b\\n" "${COL_LIGHT_RED}" "${piholeGitUrl}" "${PI_HOLE_LOCAL_REPO}" "${COL_NC}"; \
         exit 1; \
         }
+        fetch_checkout_pull_branch ${PI_HOLE_LOCAL_REPO} alpine
```
Comment (ilpianista) [inline automated install/basic-install.sh:2230]: To be removed before merging.

**Comment by auyer** [comment]
Is this PR active ? There is no way Travis is waiting since May 26th.
I've tested the script in Alpine 3.12 AARCH64, and got either a `[✗] OS distribution not supported` when using without sudo, or a `Cancel was selected, exiting installer` when running with sudo.

I'm really looking forward to alpine support. I've tried installing it in a couple methods and failed.

**Comment by ilpianista** [comment]
> Is this PR active ? There is no way Travis is waiting since May 26th.

I think upstream may not be interested on this since I didn't get any reply.

In any case, I'll rebase this and try to install it again from scratch to test it again.

**Comment by deftdawg** [comment]
If you need something portable that runs nicely on alpine, you might look into adguard-home.  Being a statically linked Go binary, I even have it running on an arm nased router.

**Comment by PromoFaux** [comment]
The issue with adding additional operating systems is one of support. None of the team uses Alpine Linux, which leaves us in a tricky position troubleshooting issues that an Alpine user may have.

Perhaps you could fork and maintain an Alpine version of the project, in much the same way that was done for Arch via the AUR (https://aur.archlinux.org/packages/pi-hole-server/) (CC: @graysky2 / @max72bra)

There is, of course, always https://github.com/pi-hole/docker-pi-hole

---

Side note: as development progresses with `FTL`, we actually are moving closer to the idea of a single binary, with a minimal bootstrap script to get things installed. But that is currently not on the immediate horizon.

**Comment by graysky2** [comment]
@max72bra is really the Arch guy for Pi-hole.  I just helped out with some trivial packaging stuff in the beginning.

**Comment by ilpianista** [comment]
> The issue with adding additional operating systems is one of support. None of the team uses Alpine Linux, which leaves us in a tricky position troubleshooting issues that an Alpine user may have.

Thank you for your reply. I'll close this PR, but I'll keep my repo alive.

> Side note: as development progresses with `FTL`, we actually are moving closer to the idea of a single binary, with a minimal bootstrap script to get things installed. But that is currently not on the immediate horizon.

Fingers crossed then! :crossed_fingers:

**Comment by dschaper** [comment]
Compiling FTL for `musl` is not trivial, and once you modify Alpine to run `glibc` then you might as well just run Debian.

**Pull request closed without merging**